K8ssandraTask test fail in CI

**What happened?**

The K8ssandraTask controller tests fail intermittently:

```
2023-02-06T23:12:40.4657128Z 2023-02-06T23:12:33.024Z	INFO	control/k8ssandratask_controller.go:85	Fetching task	{"controller": "k8ssandratask", "controllerGroup": "control.k8ssandra.io", "controllerKind": "K8ssandraTask", "K8ssandraTask": {"name":"upgradesstables","namespace":"ns-nv4vvzx9r"}, "namespace": "ns-nv4vvzx9r", "name": "upgradesstables", "reconcileID": "73653ed7-cad8-46d9-89c1-fa277fa3e860", "K8ssandraTask": "ns-nv4vvzx9r/upgradesstables"}
2023-02-06T23:12:40.4658992Z 2023-02-06T23:12:33.029Z	DEBUG	events	recorder/recorder.go:103	Created CassandraTask ns-nv4vvzx9r.upgradesstables-dc1 in context cluster-1-4mzpch	{"type": "Normal", "object": {"kind":"K8ssandraTask","namespace":"ns-nv4vvzx9r","name":"upgradesstables","uid":"a3df484a-4b97-42b7-a739-f4ef096c51ad","apiVersion":"control.k8ssandra.io/v1alpha1","resourceVersion":"305"}, "reason": "CreateCassandraTask"}
2023-02-06T23:12:40.4660830Z 2023-02-06T23:12:33.034Z	DEBUG	events	recorder/recorder.go:103	Created CassandraTask ns-nv4vvzx9r.upgradesstables-dc2 in context cluster-2-k2mppr	{"type": "Normal", "object": {"kind":"K8ssandraTask","namespace":"ns-nv4vvzx9r","name":"upgradesstables","uid":"a3df484a-4b97-42b7-a739-f4ef096c51ad","apiVersion":"control.k8ssandra.io/v1alpha1","resourceVersion":"305"}, "reason": "CreateCassandraTask"}
2023-02-06T23:12:40.4662612Z 2023-02-06T23:12:33.038Z	INFO	control/k8ssandratask_controller.go:85	Fetching task	{"controller": "k8ssandratask", "controllerGroup": "control.k8ssandra.io", "controllerKind": "K8ssandraTask", "K8ssandraTask": {"name":"upgradesstables","namespace":"ns-nv4vvzx9r"}, "namespace": "ns-nv4vvzx9r", "name": "upgradesstables", "reconcileID": "df107eb9-bf9f-44c7-b8c5-1bbddd136e4d", "K8ssandraTask": "ns-nv4vvzx9r/upgradesstables"}
2023-02-06T23:12:40.4664508Z 2023-02-06T23:12:33.039Z	INFO	control/k8ssandratask_controller.go:175	CassandraTask already exists, retrieving its status	{"controller": "k8ssandratask", "controllerGroup": "control.k8ssandra.io", "controllerKind": "K8ssandraTask", "K8ssandraTask": {"name":"upgradesstables","namespace":"ns-nv4vvzx9r"}, "namespace": "ns-nv4vvzx9r", "name": "upgradesstables", "reconcileID": "df107eb9-bf9f-44c7-b8c5-1bbddd136e4d", "K8ssandraTask": "ns-nv4vvzx9r/upgradesstables", "CassandraTask": "ns-nv4vvzx9r/upgradesstables-dc1"}
2023-02-06T23:12:40.4666514Z 2023-02-06T23:12:33.041Z	INFO	control/k8ssandratask_controller.go:175	CassandraTask already exists, retrieving its status	{"controller": "k8ssandratask", "controllerGroup": "control.k8ssandra.io", "controllerKind": "K8ssandraTask", "K8ssandraTask": {"name":"upgradesstables","namespace":"ns-nv4vvzx9r"}, "namespace": "ns-nv4vvzx9r", "name": "upgradesstables", "reconcileID": "df107eb9-bf9f-44c7-b8c5-1bbddd136e4d", "K8ssandraTask": "ns-nv4vvzx9r/upgradesstables", "CassandraTask": "ns-nv4vvzx9r/upgradesstables-dc2"}
2023-02-06T23:12:40.4668904Z 2023-02-06T23:12:33.043Z	ERROR	controller/controller.go:326	Reconciler error	{"controller": "k8ssandratask", "controllerGroup": "control.k8ssandra.io", "controllerKind": "K8ssandraTask", "K8ssandraTask": {"name":"upgradesstables","namespace":"ns-nv4vvzx9r"}, "namespace": "ns-nv4vvzx9r", "name": "upgradesstables", "reconcileID": "df107eb9-bf9f-44c7-b8c5-1bbddd136e4d", "error": "Operation cannot be fulfilled on k8ssandratasks.control.k8ssandra.io \"upgradesstables\": the object has been modified; please apply your changes to the latest version and try again"}
2023-02-06T23:12:40.4670025Z sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
2023-02-06T23:12:40.4671016Z 	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:326
2023-02-06T23:12:40.4671640Z sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
2023-02-06T23:12:40.4672335Z 	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:273
2023-02-06T23:12:40.4672957Z sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
2023-02-06T23:12:40.4673681Z 	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:234

[...]

2023-02-06T23:12:40.4925164Z --- FAIL: TestK8ssandraTask (56.53s)
2023-02-06T23:12:40.4925787Z     --- FAIL: TestK8ssandraTask/ExecuteSequentialK8ssandraTask (5.02s)
2023-02-06T23:12:40.4926431Z         k8ssandratask_controller_test.go:180: Create a K8ssandraTask
2023-02-06T23:12:40.4927090Z         k8ssandratask_controller_test.go:204: Check that CassandraTask 1 has been created
2023-02-06T23:12:40.4927664Z         k8ssandratask_controller_test.go:422: 
2023-02-06T23:12:40.4928736Z             	Error Trace:	/home/runner/work/k8ssandra-operator/k8ssandra-operator/controllers/control/k8ssandratask_controller_test.go:422
2023-02-06T23:12:40.4930479Z             	            				/home/runner/work/k8ssandra-operator/k8ssandra-operator/controllers/control/k8ssandratask_controller_test.go:205
2023-02-06T23:12:40.4944805Z             	            				/home/runner/work/k8ssandra-operator/k8ssandra-operator/controllers/control/testenv.go:307
2023-02-06T23:12:40.4945662Z             	Error:      	Condition never satisfied
2023-02-06T23:12:40.4946388Z             	Test:       	TestK8ssandraTask/ExecuteSequentialK8ssandraTask
```

--- a/CHANGELOG/CHANGELOG-1.6.md
+++ b/CHANGELOG/CHANGELOG-1.6.md
@@ -17,3 +17,4 @@ When cutting a new release, update the `unreleased` heading to the tag being gen
 
 * [CHANGE] [#848](https://github.com/k8ssandra/k8ssandra-operator/issues/848) Perform Helm releases in the release workflow
 * [FEATURE] [#815](https://github.com/k8ssandra/k8ssandra-operator/issues/815) Add configuration block to CRDs for new Cassandra metrics agent.
+* [BUGFIX] [#854](https://github.com/k8ssandra/k8ssandra-operator/issues/854) Use Patch() to update K8ssandraTask status.

--- a/controllers/control/k8ssandratask_controller.go
+++ b/controllers/control/k8ssandratask_controller.go
@@ -181,8 +181,9 @@ func (r *K8ssandraTaskReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 				break
 			}
 		}
+		patch := client.MergeFrom(kTask.DeepCopy())
 		kTask.RefreshGlobalStatus(len(dcs))
-		if err = r.Status().Update(ctx, kTask); err != nil {
+		if err = r.Status().Patch(ctx, kTask, patch); err != nil {
 			return ctrl.Result{}, err
 		}
 		// If the status update set a completion time, we want to reconcile again in order to handle the TTL. But

